### ISSUE TITLE
CMake complains about cyclic dependencies

### ISSUE THREAD

**Issue by erAck** [codilime/veles#309]
I did a `git checkout 2017.06.0.SWOND`
`cmake -D CMAKE_BUILD_TYPE=Release ..`
gives
```
CMake Error: The inter-target dependency graph contains the following strongly connected component (cycle):
  "parser" of type STATIC_LIBRARY
    depends on "veles_db" (weak)
    depends on "parser_autogen" (strong)
  "veles_db" of type STATIC_LIBRARY
    depends on "parser" (weak)
    depends on "veles_db_autogen" (strong)
  "parser_autogen" of type UTILITY
    depends on "veles_db" (strong)
  "veles_db_autogen" of type UTILITY
    depends on "parser" (strong)
At least one of these targets is not a STATIC_LIBRARY.  Cyclic dependencies are allowed only among static libraries.
```

**Comment by mkow** [comment]
This looks like a duplicate of #290. @koriakin: Could you please take a look at this?

@erAck We'll ping you when this get fixed, for now you can try using older CMake version (3.8 should work AFAIK). Sorry for problems ;)

**Comment by erAck** [comment]
Indeed, using cmake 3.9.0, sorry for duplicate.

**Comment by koriakin** [comment]
@erAck This has been fixed in current master.

**Comment by erAck** [comment]
I cherry-picked commit 7d973b6779374d45dc415fd24e27652aaa0b8f85 from origin/wip/mwk/cmake-fix onto 2017.06.0.SWOND and can confirm the build now completes. Thanks.
